### PR TITLE
feat(ui): operator transmit timer in the status bar (MOX/PTT/VOX, not TCI/DAX)

### LINKED ISSUE #4127: visible tx timer
### Request preparation

- [ ] I used an AI assistant to help structure this request
- [ ] I checked for existing issues covering the same feature

### What would you like?

please include the option to turn on a self resetting tx timer, that has a visual display how many minutes and seconds you have been talking per mox cycle. this will help the long winded people know when to stop as is for myself. other possible features would be a beep that could be settable to a certain number of seconds into a tx, but not interrupt like tot does.

### PR DESCRIPTION
**Pull request by jensenpat**
## Summary

Adds a **transmit timer** to the top status bar, immediately to the left of the **PC Audio** button. It is a bright-green, square-outlined elapsed-time readout that runs **only for operator-driven transmits** — MOX, local/hardware PTT, footswitch, VOX, CW, tune — and **never** for TCI-hardware or DAX (external-app) transmits.

State model (as requested in the design discussion):

- **Idle → not displayed.** No box, no placeholder — the status bar collapses cleanly.
- **Key up → appears and counts up** from `0:00` (MM:SS, rolling to `H:MM:SS` past an hour). Each transmission is its own timer starting at zero.
- **Unkey → holds** the final elapsed reading for **15 s**, then **fades out** over 800 ms.

Colour is `#20c060` — the same network-status / heartbeat green already used in the title bar.

### Running (1 W dummy load, ANT2)
running

### Held after unkey (frozen at final elapsed, before fade)
hold

### Idle (hidden)
idle

## Why the TCI/DAX exclusion is done at the keying funnel

The FlexRadio interlock reports both a local software MOX **and** a TCI-hardware/DAX key as `source=SW`, so the two owned software paths are indistinguishable at the interlock level. The initiating source is therefore captured where keying is initiated:

- `TransmitModel::activePttSource()` is set by `requestPttOn()` and `RadioModel::setTransmit()`.
- It is reset to `Mox` on full unkey, so a subsequent hardware-mic or VOX key — which never carries a source through those entry points — still counts as an operator transmit.

`RadioModel::operatorTransmitChanged(bool)` emits the gated edge, driven off `TransmitModel::transmittingChanged` (which the interlock handler already forces false for DAX and other-client TX). The pure predicate is extracted as **`RadioStatusOwnership::operatorTransmitActive`** so the exclusion is unit-tested — the negative (TCI/DAX) paths are hard to drive on live hardware.

## Testing

Proven live on a **FLEX-8400M** via the agent automation bridge, 1 W into a dummy load on **ANT2**:

- Hidden when idle; appears + `running=true` within 0.3 s of a MOX key.
- Displayed seconds track true elapsed within ~200 ms (5 Hz repaint).
- Freezes + `holding=true` on unkey; fades to `visible=false` after the 15 s hold.
- New **`get txtimer`** bridge verb exposes `visible/running/holding/fading/elapsedMs/text/opacity` for assertions.
- `radio_status_ownership_test` gains six operator-transmit-gate cases (MOX runs it; TCI-hardware, DAX-by-source, DAX-by-active-guard, and the optimistic DAX edge all stay excluded) — all pass.

## New automation-bridge verb

`get txtimer` — documented in `docs/automation-bridge.md`. Suggested follow-on verbs for richer TX-indicator testing are noted in the PR discussion.

## Files

- **Model:** `TransmitModel` (source tracking), `RadioModel` (`operatorTransmitChanged` + predicate), `RadioStatusOwnership` (pure `operatorTransmitActive`).
- **UI:** `TitleBar` (the timer widget + hold/fade), wired in `MainWindow`.
- **Bridge:** `AutomationServer` `get txtimer`, `MainWindow::automationTxTimerSnapshot`, `main.cpp` handler.
- **Tests/docs:** `radio_status_ownership_test`, `docs/automation-bridge.md`.

💻 Generated with Claude Code (claude-fable-5 7/1/26) with architecture by @jensenpat

### CODE REVIEW

File: src/gui/TitleBar.cpp [hunk 1]
```diff
@@ -253,6 +256,59 @@ TitleBar::TitleBar(QWidget* parent)
...
+    markDragHandle(m_txTimerLabel);
+    // Opacity effect powers the 15s-hold → fade-out on unkey.
+    m_txTimerOpacity = new QGraphicsOpacityEffect(m_txTimerLabel);
+    m_txTimerOpacity->setOpacity(1.0);
+    m_txTimerLabel->setGraphicsEffect(m_txTimerOpacity);
+    m_txTimerLabel->setVisible(false);   // idle: not displayed
+    m_hbox->addWidget(m_txTimerLabel);
+    m_hbox->addSpacing(6);
+
+    // 1s tick that repaints the elapsed time while keyed.
```
[the remark this answers is not in the document]
Comment (jensenpat) [reply src/gui/TitleBar.cpp]: Fixed in fba31f0b — both comments now read "5 Hz" (TitleBar.cpp) and "5 Hz cadence" (TitleBar.h) to match the 200 ms `setInterval`.

File: src/gui/TitleBar.cpp [hunk 2]
```diff
@@ -873,6 +929,73 @@ void TitleBar::setOtherClientTx(bool transmitting, const QString& station)
...
+void TitleBar::updateTxTimerText()
+{
+    const qint64 ms = m_txTimerRunning ? m_txElapsed.elapsed() : m_txFrozenMs;
+    m_txTimerLabel->setText(formatTxElapsed(ms));
+}
+
+void TitleBar::setOperatorTransmitting(bool active)
+{
+    if (active) {
+        // Key-up: cancel any pending hold/fade, restart from 0:00 (each
```
[the remark this answers is not in the document]
Comment (jensenpat) [reply src/gui/TitleBar.cpp:918]: Good catch — confirmed on a FLEX-8400M (4.2.18) that break-in/QSK CW does toggle the interlock per element, so the timer thrashed back to 0:00 on every dit/dah. Rather than debounce, we decided a wall-clock over-timer just isn't the right readout for CW, so CW is now excluded entirely (fba31f0b): the pure predicate `operatorTransmitActive` gains a `modeIsCw` arg, `RadioModel::updateOperatorTransmit` reads the TX slice mode (any `CW*`), and there's a new unit-test case. Proven live on 1 W into an ANT2 dummy load — in CW the timer stays hidden through a full key/unkey cycle; USB on the same key still shows and counts.

**Review by ten9876 — DISMISSED** [review]
**Approve.** Reviewed with a multi-agent pass (model logic, TitleBar widget, bridge threading + cleanup) and applied the fixes for the edge cases that pass turned up. The core architecture was clean; these close the corners.

## Verified sound (no change needed)
- `get txtimer` threading — AutomationServer runs on the GUI thread (the `invokeMethod` used elsewhere is for the EB1 engine boundary, not thread-hopping), so reading TitleBar state directly is safe.
- Teardown order, doc/key parity, disconnect-while-keyed reset, and the TitleBar hold/fade state machine (Qt's `stop()` doesn't emit `finished()`, so no stale hide) all check out.
- TCI-hardware / DAX exclusion works for the common keying path (`requestPttOn`/`setTransmit` tag the source before the transmit edge).

## Fixes applied on this branch (d00af87)
1. **Tune source tag** — `startTune`/`startTwoToneTune` now record `m_activePttSource`, so a TCI/DAX-initiated tune (`startTune(Dax)` from `TciProtocol`) is excluded instead of inheriting the stale `Mox` tag and wrongly running the operator timer. Operator tune (source=Tune) still shows.
2. **Mid-over mode edges** — recompute on `TransmitModel::txSliceModeChanged`, so a phone↔CW switch mid-transmit (and the mode first resolving after connect) re-evaluates the CW exclusion; `transmittingChanged` alone couldn't catch it. Idempotent.
3. **DAX toggle** — `updateOperatorTransmit()` when `m_daxTxActive` changes, keeping the gate fresh on DAX TX start/stop.
4. **TitleBar idempotency** — a duplicate `operatorTransmitChanged(true)` no longer resets the elapsed clock mid-over (mirrors the unkey guard).
5. **Bridge consistency** — `txTimerState()` derives `text` from the same live `ms` as `elapsedMs`, so `get txtimer` never reports disagreeing values.

Build clean; `radio_status_ownership_test` passes.

## Left as-is (deliberate)
- **Show/hide layout twitch** — the obvious fix (`retainSizeWhenHidden`) would leave a permanent 52 px gap when idle, worse than a twitch on a deliberate key/unkey.
- **Hardcoded `#20c060`** — matches the sibling multi-Flex button's status-widget color; per AGENTS.md I don't autonomously change visual design colors (your call).
- **a11y show/hide announcement + model-level tests for the reset-to-Mox path** — nice-to-haves; the pure predicate is already unit-tested and the PR ships proof screenshots.

Nice feature — the WSJT-X-style over timer is a genuinely useful operator readout.

**Review by ten9876 — APPROVED** [review]
Approving at `4dddc072` (the conflict-resolution rebase I pushed earlier — Pan-Lock context dropped per #4116's removal, TX-timer block kept in its place, `txtimer` handler unions applied; authorship preserved, all commits signed).

Verification for this approval:
- **TX-safety surface**: the diff is purely observational — zero radio commands added (grep-verified: no `sendCommand`/keying paths); the `txtimer` bridge verb is a read-only snapshot.
- **The operator-vs-external discriminator** (`operatorTransmitActive`) is a pure predicate with unit coverage, and the reasoning holds: `isTransmitting()` already excludes other-client TX via the interlock handler; the radio's `source=SW` ambiguity between TCI-hardware and DAX is disambiguated from the remembered `PttSource`; `daxTxActive` belt-and-braces the optimistic DAX edge; and the CW exclusion is right — break-in toggles the interlock per element, which would restart a wall-clock timer every dit. The prior review round's fix (tagging `m_activePttSource` in `startTune`/`startTwoToneTune` so an external-app tune doesn't inherit a stale Mox tag) closes the one leak I'd have flagged.
- Built clean, **79/79 tests** (including the new ownership cases), CI green on the rebased head.

Scope note: this PR already carried a full review round (the edge-cases commit); this pass verified the rebase, the TX-safety surface, and the discriminator rather than repeating the 8-angle sweep.

73 — reviewed with Claude Code on behalf of KK7GWY.
